packages/web-components/fast-foundation/src/number-field/number-field.ts

```diff
@@ -0,0 +1,246 @@
+                this.min = this.max;
+                this.max = numb;
+            } else {
+                this.min = numb;
+            }
+        }
+    }
+
+    /**
+     * @internal
+     */
+    @observable
+    public defaultSlottedNodes: Node[];
+
+    /**
+     * A reference to the internal input element
+     * @internal
+     */
+    public control: HTMLInputElement;
+
+    /**
+     *
+     * @param previousValue - previous stored value
+     * @param nextValue - value being updated
+     */
+    public valueChanged(previousValue, nextValue): void {
+        super.valueChanged(previousValue, nextValue);
+
+        const numb = parseFloat(nextValue);
+        let out: number | string = numb == nextValue ? nextValue : numb;
+
+        if (nextValue === "" || isNaN(numb)) {
+            out = "";
+        } else if (this.min !== undefined && numb < this.min) {
+            out = this.min;
+        } else if (this.max !== undefined && numb > this.max) {
+            out = this.max;
+        }
+
+        this.value = out.toString();
+
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.value = this.value;
+        }
+    }
+
+    /**
+     * Increments the value using the step value
+     */
+    public stepUp(): void {
+        const steppedValue = parseFloat(this.value) + this.step;
+        const nextValue =
+            this.max !== undefined && steppedValue > this.max ? this.max : steppedValue;
+        this.value = parseFloat(nextValue.toPrecision(12)).toString();
+    }
+
+    /**
+     * Decrements the value using the step value
+     */
+    public stepDown(): void {
+        const steppedValue = parseFloat(this.value) - this.step;
+        const nextValue =
+            this.min !== undefined && steppedValue < this.min ? this.min : steppedValue;
+        this.value = parseFloat(nextValue.toPrecision(12)).toString();
+    }
+
+    /**
+     * @internal
+     */
+    public connectedCallback(): void {
+        super.connectedCallback();
+
+        this.proxy.setAttribute("type", "number");
+        this.validate();
+
+        if (this.autofocus) {
+            DOM.queueUpdate(() => {
+                this.focus();
+            });
+        }
+    }
+
+    /**
+     * Handles the internal control's `input` event
+     * @internal
+     */
+    public handleTextInput(): void {
+        this.value = this.control.value;
+    }
+
+    /**
+     * Change event handler for inner control.
+     * @remarks
+     * "Change" events are not `composable` so they will not
+     * permeate the shadow DOM boundary. This fn effectively proxies
+     * the change event, emitting a `change` event whenever the internal
+     * control emits a `change` event
+     * @internal
+     */
+    public handleChange(): void {
+        this.$emit("change");
+    }
+}
+
+/**
+ * Includes ARIA states and properties relating to the ARIA textbox role
+ *
+ * @public
+ */
+export class DelegatesARIANumberbox {}
```

Comment:
I don't think there is an actual ARIA Numberbox, so I wonder if this would simply use DelegatesARIATextbox as well.

Reply:
Updated this.